client-react/src/pages/app/functions/function/function-editor/FunctionEditorCommandBar.tsx

```
@@ -1,108 +1,120 @@
+  saveFunction,
+  setSelectedFileName,
+  setShowInvalidFileSelectedWarning,
+  testDisabled,
+  testFunction,
+  testIntegrationFunction,
+  upload,
+  urlObjs,
+  runtimeVersion,
+}: FunctionEditorCommandBarProps) => {
   const { t } = useTranslation();
   const portalCommunicator = useContext(PortalContext);
   const functionEditorContext = useContext(FunctionEditorContext);
   const siteStateContext = useContext(SiteStateContext);
 
   const [isDialogVisible, setIsDialogVisible] = useState<boolean>(false);
 
-  const onClickGetFunctionUrlCommand = () => {
+  const examples = useCosmosDbExamples(functionInfo);
+
+  const testIntegrationDisabled = useMemo<boolean>(() => {
+    return !Object.values(examples).some(value => !!value);
+  }, [examples]);
+
+  const onClickGetFunctionUrlCommand = useCallback(() => {
     resetInvalidFileSelectedWarningAndFileName();
     setIsDialogVisible(true);
-  };
+  }, [resetInvalidFileSelectedWarningAndFileName]);
 
   const getFunctionUrlButtonRef = useRef<IContextualMenuRenderItem | null>(null);
   const uploadFileRef = useRef<HTMLInputElement | null>(null);
 
-  const onUploadButtonClick = () => {
-    if (uploadFileRef && uploadFileRef.current) {
-      uploadFileRef.current.click();
-    }
-  };
+  const onUploadButtonClick = useCallback(() => {
+    uploadFileRef.current?.click();
+  }, []);
 
-  const uploadFile = e => {
-    const file = e.target && e.target.files && e.target.files.length > 0 && e.target.files[0];
-    const isValidFile = !!file && !!file.size;
-    if (isValidFile) {
-      upload(file);
-    }
+  const uploadFile = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const file = e.target.files?.[0];
+      const isValidFile = file?.size ?? 0 > 0;
+      if (isValidFile) {
+        upload(file);
+      }
 
-    setSelectedFileName((!!file && file.name) || '');
-    setShowInvalidFileSelectedWarning(!isValidFile);
-  };
+      setSelectedFileName(file?.name ?? '');
```

we don't want to add an empty name here right? Can we default to some random Guid instead?

---

`file?.name` is only undefined if no file was uploaded so it doesn't matter what it's named.